docs/develop/python/message-passing.mdx

```diff
@@ -27,155 +27,171 @@ tags:
+  The validator must take the same argument type as the handler, but always returns `None`.
 
-To send a Signal from the Client, use the [signal()](https://python.temporal.io/temporalio.client.WorkflowHandle.html#signal) function on the Workflow handle.
+- To reject an Update, you raise an exception (of any type) in the validator.
 
-To get the Workflow handle, you can use any of the following options.
+- Update (and Signal) handlers can also be `async def` and thus use Activities, Child Workflows, durable [`asyncio.sleep`](https://docs.python.org/3/library/asyncio-task.html#asyncio.sleep) timers, [`workflow.wait_condition`](https://python.temporal.io/temporalio.workflow.html#wait_condition) conditions, etc. See [Async handlers](#async-handlers) below, and read [Workflow message passing](/encyclopedia/workflow-message-passing) for an introduction to safe usage of async Signal and Update handlers.
 
-- Use the [get_workflow_handle()](https://python.temporal.io/temporalio.client.Client.html#get_workflow_handle) method.
-- Use the [get_workflow_handle_for()](https://python.temporal.io/temporalio.client.Client.html#get_workflow_handle_for) method to get a type-safe Workflow handle by its Workflow Id.
-- Use the [start_workflow()](https://python.temporal.io/temporalio.client.Client.html#start_workflow) to start a Workflow and return its handle.
 
-<div class="copycode-notice-container">
-  <a href="https://github.com/temporalio/documentation/blob/main/sample-apps/python/signal_your_workflow/signal_dacx.py">
-    View the source code
-  </a>{' '}
-  in the context of the rest of the application code.
-</div>
+## Sending messages
+
+To send Queries, Signals, or Updates you call methods on a [WorkflowHandle](https://python.temporal.io/temporalio.client.WorkflowHandle.html) object.
+To obtain the Workflow handle, you can:
+- Use the [start_workflow](https://python.temporal.io/temporalio.client.Client.html#start_workflow) to start a Workflow and return its handle.
+- Use the [get_workflow_handle_for](https://python.temporal.io/temporalio.client.Client.html#get_workflow_handle_for) method to get a Workflow handle.
 
+For example:
 ```python
-from temporalio.client import Client
-# ...
-# ...
-    client = await Client.connect("localhost:7233")
-    handle = await client.start_workflow(
-        GreetingWorkflow.run,
-        id="your-greeting-workflow",
-        task_queue="signal-tq",
-    )
-    await handle.signal(GreetingWorkflow.submit_greeting, "User 1")
+client = await Client.connect("localhost:7233")
+workflow_handle = await client.start_workflow(
+    GreetingWorkflow.run, id="greeting-workflow-1234", task_queue="my-task-queue"
+)
 ```
 
-### Send a Signal from a Workflow {#send-signal-from-workflow}
+To find out what argument type to provide (and what return type to expect for Queries and Updates), look at the corresponding handler method on the Workflow Definition.
```

Comment:
This is a style guide issue, imo -- rather than directing someone to "find out" and "look at" something else, this should be expressed more straightforwardly, e.g., "The argument type should match the corresponding handler method... for example, X, Y."

Reply:
Thanks, I don't disagree with your edit. FWIW what I was going for here is the following: I was trying to write this from the point of view of someone writing client-side code who needs to switch attention to the Workflow definition (maybe using IDE hover-over / command-click) to look up the signature of the handler method. But I can see that it's either more conversational/informal thn we want, or not actually more helpful than the more direct phrasing, or both!